# 📊 Medidas de tendencia central

## 💼 Salarios de empleados

📥 **Importamos la librería pandas** para trabajar con datos en forma de tabla (DataFrames).

In [1]:
import pandas as pd

📂 **Cargamos los datos de salarios** desde un archivo CSV que contiene la información de los empleados.

In [2]:
salarios = pd.read_csv("data/salarios_empleados.csv")

🔍 **Mostramos la tabla de salarios** para revisar los datos.

In [3]:
salarios

,nombre,salario
0,Juan,25000
1,Pancho,5000
2,Luis,3100
3,Paco,4200
4,Carolina,1500
5,Rosa,2800


### 📐 Obtener la media

📊 **Calculamos la media (promedio) de los salarios.**
La media representa el valor promedio de todos los datos. `numeric_only=True` ignora columnas no numéricas.

In [4]:
salarios.mean(numeric_only=True)

salario    6933.333333
dtype: float64

📈 La media es `6933.333333`

### 📏 Obtener la mediana

🔸 **La mediana es el valor central** cuando los datos están ordenados. Es útil cuando hay valores extremos (muy altos o muy bajos).

In [5]:
salarios.median(numeric_only=True)

salario    3650.0
dtype: float64

📏 La mediana es `3650.0`

### 🔁 Obtener la moda

📌 **La moda es el valor que más se repite.** Si no hay un único valor más frecuente, se muestran todos los posibles.

In [6]:
salarios.mode(numeric_only=True)

,salario
0,1500
1,2800
2,3100
3,4200
4,5000
5,25000


❗ No hay un valor único que se repita más veces. Se muestran todos los valores posibles.

## 🧑‍🏫 Asistencias de un grupo en una semana

📥 **Leemos los datos de asistencias** en la semana desde un archivo CSV.

In [7]:
asistencias = pd.read_csv("data/asistencias_clases.csv")
asistencias

,dia,asistencia
0,lunes,6
1,martes,3
2,miercoles,8
3,jueves,6
4,viernes,4


### 📐 Obtener la media

📊 **Calculamos el promedio de asistencias** del grupo a lo largo de la semana.

In [8]:
asistencias.mean(numeric_only=True)

asistencia    5.4
dtype: float64

📈 El promedio de asistencias es de `5.4`

### 📏 Obtener la mediana

🔸 **La mediana** nos da el número de asistencias que está justo en el medio del grupo de datos ordenados.

In [9]:
asistencias.median(numeric_only=True)

asistencia    6.0
dtype: float64

📏 La mediana es de `6.0`

### 🔁 Obtener la moda

📌 **Calculamos la moda** para ver cuál es el número de asistencias más común entre los estudiantes.

In [10]:
asistencias.mode(numeric_only=True)

,asistencia
0,6


🔁 La moda es `6`

✨ *(Puedes agregar más análisis, como gráficos de barras o histogramas para visualizar los datos.)*